base/src/types.rs

```diff
@@ -750,23 +760,31 @@ impl<'a, I, T, E> DisplayType<'a, I, T, E>
                     }
                 }
             }
-            Type::Variants(ref variants) => {
+            Type::Variant(ref row) => {
                 let mut first = true;
                 let mut doc = arena.nil();
-                for variant in variants {
-                    if !first {
-                        doc = doc.append(arena.newline());
-                    }
-                    first = false;
-                    doc = doc.append("| ")
-                        .append(variant.0.as_ref());
-                    for arg in arg_iter(&variant.1) {
-                        doc = chain![arena;
-                                     doc,
-                                     " ",
-                                     dt(self.env, Prec::Constructor, &arg).pretty(arena)];
+
+                match **row {
+                    Type::EmptyRow => (),
+                    Type::ExtendRow { ref fields, .. } => {
+                        for field in fields.iter() {
+                            if !first {
+                                doc = doc.append(arena.newline());
+                            }
+                            first = false;
+                            doc = doc.append("| ")
+                                .append(field.name.as_ref());
+                            for arg in arg_iter(&field.typ) {
+                                doc = chain![arena;
+                                            doc,
+                                            " ",
+                                            dt(self.env, Prec::Constructor, &arg).pretty(arena)];
+                            }
+                        }
                     }
-                }
+                    _ => panic!("Polymorphic variant syntax not yet decided"),
```

Comment:
"Unexpected type {} in variant" or something? May be a bit clearer about what happened if this is actually reached.